### PR TITLE
Features branch

### PR DESCRIPTION
**Pull request by FrancoRosa**
We kindly request for reviewing our Micro-Reddit project.
In this project we managed to work with Active Record, and we did the following:
 - Create the User model.
 - Create the Post model.
 - Create the Comments model.
 - Make the relationships and validations required.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+
+  validates :title, :body, :user_id, presence: true
```
Comment (stratospherique) [inline app/models/post.rb]: * [x] You shouldn't validate the presence of a Foreign key. In usual cases, associations will do that for you :gear:

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  has_many :posts
+
```
Comment (stratospherique) [inline app/models/user.rb]: :warning:  `has_many :comments` is missing.

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  email_validor = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  validates :username, :email, :password, presence: true
+  validates :username, :email, uniqueness: true
+  validates :username, length: { in: 4..12 }
+  validates :password, length: { in: 6..16 }
+  validates :email,
+            format: {
+              with: email_validor,
+              message: 'this is not a valid email'
+            }
```
Comment (stratospherique) [inline app/models/user.rb:14]: - **[SUGGESTION]** Validating each column separately in one line is much better and easier to read, but I understand if you are facing issues with rubocop.

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great start @team :clap: :clap: However, some issues related to the models need to be fixed. Check :arrow_down: 

##### User

* [x] `has_many :comments` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/user_has_many_comments.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/building-with-active-record-ruby-on-rails))

- **[REQUIRED]** Check the code comments below, I noted some non-recommended practices that were incorporated in your code.

### That's it :relieved: When finished, kindly submit a code re-review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

File: app/models/user.rb (hunk 3 above)
Comment (FrancoRosa) [reply app/models/user.rb:14]: **Thank you for Ahmed your review!** 
It was helpful, we also added the **dependent: :destroy** in the **has_many** method to be able to delete post and comments related to one user.

**Review by Torres-ssf — APPROVED** [review]
### Project Approved :+1: :+1: :+1: 
Hi there team :hand: 
Great job on this project :clap:, all required changes were implemented :smiley: 
Good luck with the next one

Happy coding :computer: 
**Sergio Torres**

**Pull request merged**